orix/tests/quaternion/test_orientation.py

```diff
@@ -168,22 +177,89 @@ def test_transpose_3d(shape, expected_shape, axes):
 
 
 def test_transpose_symmetry():
-    o1 = Orientation.random_vonmises((11, 3)).set_symmetry(Oh)
+    o1 = Orientation.random_vonmises((11, 3))
+    o1.symmetry = Oh
+    o1 = o1.compute_symmetry_reduced_orientations()
     o2 = o1.transpose()
 
     assert o1.symmetry == o2.symmetry
 
 
+def test_symmetry_property_orientation():
+    o = Orientation.random((3, 2))
+
+    sym = Oh
+    o.symmetry = sym
+
+    assert o.symmetry == sym
+    assert o._symmetry == (C1, sym)
+
+
+def test_symmetry_property_orientation_data():
+    """Test that data remains unchanged after setting symmetry property."""
+    o = Orientation.random((3, 2))
+    d1 = o.data.copy()
+    o.symmetry = Oh
+
+    assert np.allclose(o.data, d1)
+
+
+def test_symmetry_property_misorientation():
+    o = Misorientation.random((3, 2))
+    o.symmetry = (Oh, C3)
+
+    assert o.symmetry == (Oh, C3)
+    assert o._symmetry == (Oh, C3)
+
+
+def test_symmetry_property_wrong_type_orientation():
+    o = Orientation.random((3, 2))
+
+    with pytest.raises(TypeError, match="Value must be an instance of"):
+        o.symmetry = 1
+
+
+def test_symmetry_property_wrong_type_misorientation():
+    o = Misorientation.random((3, 2))
+
+    with pytest.raises(ValueError, match="Values must be a tuple"):
+        # not Symmetry
+        o.symmetry = (1, 2)
+
+    with pytest.raises(ValueError, match="Values must be a tuple"):
+        # not Symmetry
+        o.symmetry = (C1, 2)
+
+    with pytest.raises(TypeError, match="Values must be a tuple"):
+        # not list or tuple
+        o.symmetry = 1
```

Comment:
Shorten with `pytest.mark.parametrize`?

```suggestion
@pytest.mark.parametrize("error_type, value", [(ValueError, (1, 2)), (ValueError, (C1, 2)), (TypeError, 1)])
def test_symmetry_property_wrong_type_misorientation(error_type, value):
    mori = Misorientation.random((3, 2))
    with pytest.raises(error_type, match="Values must be a tuple"):
        mori.symmetry = value
```